From 13f7974513468f3f7b6d911dec9336c8de719ca1 Mon Sep 17 00:00:00 2001
From: Jack Yu <jackcwyu@google.com>
Date: Wed, 30 Jan 2019 14:59:09 +0800
Subject: [PATCH 4/5] Prevent OOB error in rw_i93_sm_read_ndef()

Bug: 122035770
Test: NFC tag reading
Change-Id: Ic006676962578871ad7ce20064226b096aad662f
Merged-In: I053e9f01dc921bab55b5781c83f048d2638d5b87
(cherry picked from commit 61fcf8972501f9897e9e6e0bc37c60e908421e9a)
---
 src/nfc/tags/rw_i93.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/nfc/tags/rw_i93.c b/src/nfc/tags/rw_i93.c
index 605ce1b..3b4e253 100644
--- a/src/nfc/tags/rw_i93.c
+++ b/src/nfc/tags/rw_i93.c
@@ -1994,6 +1994,12 @@ void rw_i93_sm_read_ndef (BT_HDR *p_resp)
 
     RW_TRACE_DEBUG0 ("rw_i93_sm_read_ndef ()");
 
+    if (length == 0) {
+        android_errorWriteLog(0x534e4554, "122035770");
+        rw_i93_handle_error(NFC_STATUS_FAILED);
+        return;
+    }
+
     STREAM_TO_UINT8 (flags, p);
     length--;
 
-- 
2.7.4

